hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -1970,18 +1970,18 @@ class IRSuite extends HailSuite {
       Array(0, 1, 2, 15, 16),
       Array(3, 4, 5, 17, 18))
 
-    assertNDEvals(NDArrayConcat(nds(nd1, rowwise), 0), rowwiseExpected)
-    assertNDEvals(NDArrayConcat(nds(nd1, rowwise, emptyRowwise), 0), rowwiseExpected)
-    assertNDEvals(NDArrayConcat(nds(nd1, emptyRowwise, rowwise), 0), rowwiseExpected)
+    assertNDEvals(TInt32(), NDArrayConcat(nds(nd1, rowwise), 0), rowwiseExpected)
+    assertNDEvals(TInt32(), NDArrayConcat(nds(nd1, rowwise, emptyRowwise), 0), rowwiseExpected)
+    assertNDEvals(TInt32(), NDArrayConcat(nds(nd1, emptyRowwise, rowwise), 0), rowwiseExpected)
 
-    assertNDEvals(NDArrayConcat(nds(nd1, colwise), 1), colwiseExpected)
-    assertNDEvals(NDArrayConcat(nds(nd1, colwise, emptyColwise), 1), colwiseExpected)
-    assertNDEvals(NDArrayConcat(nds(nd1, emptyColwise, colwise), 1), colwiseExpected)
+    assertNDEvals(TInt32(), NDArrayConcat(nds(nd1, colwise), 1), colwiseExpected)
+    assertNDEvals(TInt32(), NDArrayConcat(nds(nd1, colwise, emptyColwise), 1), colwiseExpected)
+    assertNDEvals(TInt32(), NDArrayConcat(nds(nd1, emptyColwise, colwise), 1), colwiseExpected)
 
     // FIXME: This is changing type during PruneDeadFields for some reason...
-//    assertNDEvals(NDArrayConcat(nds(nd1, na), 1), null)
-    assertNDEvals(NDArrayConcat(nds(na, na), 1), null)
-    assertNDEvals(NDArrayConcat(NA(TArray(TNDArray(TInt32(), Nat(2)))), 1), null)
+//    assertNDEvals(TInt32(), NDArrayConcat(nds(nd1, na), 1), null)
```

Comment:
does this work if it's uncommented now?

Reply:
no :(

It's an issue where PruneDeadFields changes ndarray's element type to +TInt32 somehow, which actually seems totally fine except for the fact that we're still storing requiredness on the virtual type. Hopefully once that's gone I can uncomment this.